### PR TITLE
[Documentations] Remove Site and Region FKs from model classes and enforce Location FK not null.

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: #2957
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/docs/release-notes/version-2.0.md [hunk 1]
```diff
@@ -78,6 +82,16 @@ The `Site` and `Region` models have been removed in v2.0 and have been replaced
...
 
+##### Remove Site and Region Related Fields from Models
+
+`Region` and `Site` relationships are being removed from these models: `CircuitTermination`, `Device`, `Location`, `Rack`, `RackGroup`, `PowerFeed`, `PowerPanel`, `ConfigContext`, `Prefix`, `VLAN`, `VLANGroup`, `Cluster`.
+
+`Region` and `Site` ContentTypes are being replaced with `Location` ContentType from these models: `ComputedFields`, `CustomFields`, `CustomLinks`, `ExportTemplates`, `ImageAttachments`, `JobHooks`, `Notes`, `Relationships`, `Statuses`, `Tags` and `Webhooks`.
+
+"region" and "site" keys are being removed in the filter data of `DynamicGroups`. The old keys' associated values are being added to the existing "location" key and its associated list or to a new "location" key with an empty list if there is no existing "location" key.
+
+Checkout the API and UI endpoints changes incurred by the changes stated above in the ["Upgrading from Nautobot v1.X"](../installation/upgrading-from-nautobot-v1.md) guide.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.0.md:93]:
```suggestion
Check out the API and UI endpoints changes incurred by the changes stated above in the ["Upgrading from Nautobot v1.X"](../installation/upgrading-from-nautobot-v1.md) guide.
```

**Review by glennmatthews — APPROVED** [review]

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md [hunk 2]
```diff
@@ -57,31 +57,45 @@
...
 ### Replaced Models
 
 The `ipam.Role`, `dcim.RackRole`, and `dcim.DeviceRole` models have been removed and replaced by a single `extras.Role` model. This means that any references to the removed models in the code now use the `extras.Role` model instead.
 
-| Removed Model     | Replaced With |
-|-------------------|---------------|
-| `dcim.DeviceRole` | `extras.Role` |
-| `dcim.RackRole`   | `extras.Role` |
-| `ipam.Role`       | `extras.Role` |
+The `dcim.region` and `dcim.site` models have been removed and replaced by `dcim.location` model. This means that any references to the removed models in the code now use the `dcim.location` model instead with `LocationType` "Site" and "Region".
```
Comment (jathanism) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:92]: Capitalizing the model names for consistency.

```suggestion
The `dcim.Region` and `dcim.Site` models have been removed and replaced by `dcim.Location` model. This means that any references to the removed models in the code now use the `dcim.Location` model instead with `LocationType` "Site" and "Region".
```

File: nautobot/docs/release-notes/version-2.0.md [hunk 3]
```diff
@@ -36,6 +36,8 @@ The `Site` and `Region` models have been removed in v2.0 and have been replaced
 
     - Model instances that had a `site` field (`CircuitTermination`, `Device`, `PowerPanel`, `RackGroup`, `Rack`, `Prefix`, `VLANGroup`, `VLAN`, `Cluster`) assigned and *did not* have a `location` attribute assigned will be updated to have their `location` point to the new `Location` corresponding to that `Site`. All other attributes on these models will remain unchanged.
 
+    - Model instances that have a `site` ContentType associated with it (`ComputedFields`, `CustomFields`, `CustomLinks`, `ExportTemplates`, `ImageAttachments`, `JobHooks`, `Notes`, `Relationships`, `Statuses`, `Tags` and `Webhooks`) will have their `site` ContentType replaced by `location` ContentType and all other attributes on these models will remain unchanged.
```
Comment (jathanism) [inline nautobot/docs/release-notes/version-2.0.md:39]:
```suggestion
    - Model instances that were previously associated to the `ContentType` for `Site` (`ComputedField`, `CustomField`, `CustomLink`, `ExportTemplate`, `ImageAttachment`, `JobHook`, `Note`, `Relationship`, `Status`, `Tag` and `Webhook`) will have their `ContentType` replaced with `Location`. All other attributes on these models will remain unchanged.
```

File: nautobot/docs/release-notes/version-2.0.md [hunk 4]
```diff
@@ -58,7 +60,9 @@ The `Site` and `Region` models have been removed in v2.0 and have been replaced
...
 
-    - Model instances that had a `site` attribute (`CircuitTermination`, `Device`, `PowerPanel`, `RackGroup`, `Rack`, `Prefix`, `VLANGroup`, `VLAN`, `Cluster`) assigned and *did not* have a `location` attribute assigned will be updated to have their `location` point to the new `Location` of `LocationType` **Site**. All other attributes on these models will remain unchanged.
+    - Model instances that had a `site` attribute (`CircuitTermination`, `Device`, `Location`, `PowerPanel`, `Rack`, `RackGroup`, `Prefix`, `VLANGroup`, `VLAN`, `Cluster`) assigned and *did not* have a `location` attribute assigned will be updated to have their `location` point to the new `Location` of `LocationType` **Site**. All other attributes on these models will remain unchanged.
+
+    - Model instances that have a `region` or `site` ContentType associated with them (`ComputedFields`, `CustomFields`, `CustomLinks`, `ExportTemplates`, `ImageAttachments`, `JobHooks`, `Notes`, `Relationships`, `Statuses`, `Tags` and `Webhooks`) will have their `region` or `site` ContentType replaced by `location` ContentType and all other attributes on these models will remain unchanged.
```
Comment (jathanism) [inline nautobot/docs/release-notes/version-2.0.md:65]:
```suggestion
    - Model instances that were previously associated to the `ContentType` for `Site`  or `Region` (`ComputedField`, `CustomField`, `CustomLink`, `ExportTemplate`, `ImageAttachment`, `JobHook`, `Note`, `Relationship`, `Status`, `Tag` and `Webhook`) will have their `ContentType` replaced with `Location`. All other attributes on these models will remain unchanged.
```

File: nautobot/docs/release-notes/version-2.0.md [hunk 5]
```diff
@@ -78,6 +82,16 @@ The `Site` and `Region` models have been removed in v2.0 and have been replaced
 
     - During the data migration, **ams01-edge-01**'s `location` attribute will point to the new `Location` of `LocationType` **Site** with name **AMS01** while devices **ams01-edge-02** and **ams01-edge-03** will remain unchanged.
 
+##### Remove Site and Region Related Fields from Models
+
+`Region` and `Site` relationships are being removed from these models: `CircuitTermination`, `Device`, `Location`, `Rack`, `RackGroup`, `PowerFeed`, `PowerPanel`, `ConfigContext`, `Prefix`, `VLAN`, `VLANGroup`, `Cluster`.
+
+`Region` and `Site` ContentTypes are being replaced with `Location` ContentType from these models: `ComputedFields`, `CustomFields`, `CustomLinks`, `ExportTemplates`, `ImageAttachments`, `JobHooks`, `Notes`, `Relationships`, `Statuses`, `Tags` and `Webhooks`.
```
Comment (jathanism) [inline nautobot/docs/release-notes/version-2.0.md:89]:
```suggestion
The `ContentType` for `Region` and `Site` are being replaced with `Location` on these models: `ComputedField`, `CustomField`, `CustomLink`, `ExportTemplate`, `ImageAttachment`, `JobHook`, `Note`, `Relationship`, `Status`, `Tag` and `Webhook`.
```

File: nautobot/docs/release-notes/version-2.0.md [hunk 6]
```diff
@@ -78,6 +82,16 @@ The `Site` and `Region` models have been removed in v2.0 and have been replaced
...
+
+The "region" and "site" keys are being removed in the filter data of `DynamicGroups`. The old keys' associated values are being added to the existing "location" key and its associated list or to a new "location" key with an empty list if there is no existing "location" key.
```
Comment (jathanism) [inline nautobot/docs/release-notes/version-2.0.md:91]:
```suggestion
The `region` and `site` fields are being removed in the `filter` data of `DynamicGroup` objects. The previously associated values are being added to the existing `location` field and its associated list of filter values or to a new `location` key with an empty list if one does not exist.
```

**Review by jathanism — APPROVED** [review]
I just got some consistency tweaks for readability, but nothing showstopping!

**Pull request merged**